### PR TITLE
refactor: Remove 'util/got'

### LINKED ISSUE #6466: refactor tests to use nock instead of got mock
We should replace all `got` mocks by `nock` in tests, so #6356 would become easier

### LINKED ISSUE #6522: Remove util/got
This change:

image

Results in:

image

### PR DESCRIPTION
**Pull request by zharinov**
Closes #6466
Closes #6522

### CODE REVIEW

File: lib/manager/bazel/update.spec.ts [hunk 1]
```diff
@@ -5,7 +5,7 @@ import { UpdateType } from '../../config';
 import { updateDependency } from './update';
 
 jest.mock('hasha');
-jest.mock('../../util/got');
+jest.mock('got');
```
Comment (viceice) [inline lib/manager/bazel/update.spec.ts:8]: Why this is using got mock?

Comment (zharinov) [reply lib/manager/bazel/update.spec.ts:8]: For mocking `got.stream`, I'm figuring it out right now

File: lib/util/http/index.ts [hunk 2]
```diff
@@ -55,6 +55,7 @@ export class Http<GetOptions = HttpOptions, PostOptions = HttpPostOptions> {
     // TODO: deep merge in order to merge headers
     let options: any = {
       method: 'get',
+      retry: 0,
```
Comment (zharinov) [inline lib/util/http/index.ts:58]: Btw do we need `retry=0` for streams? @viceice

Comment (zharinov) [reply lib/util/http/index.ts:58]: According to tests, it doesn't do retry though

File: lib/manager/homebrew/update.ts [hunk 3]
```diff
@@ -181,6 +181,7 @@ export async function updateDependency({
       return fileContent;
     }
   }
+  // istanbul ignore next
```
Comment (zharinov) [inline lib/manager/homebrew/update.ts:184]: It's pretty hard to get here as soon as we've dropped `fromStream` mocking

File: lib/util/http/index.ts (hunk 2 above)
Comment (rarkins) [reply lib/util/http/index.ts:58]: We *do* want `got` to do retrying. Does this disable it?

Comment (zharinov) [reply lib/util/http/index.ts:58]: It disables it by default, in order to match current snapshots. Probably, we should adjust snapshots instead.

**Review by rarkins — CHANGES_REQUESTED** [review]
Revert `retry: 0` change

File: lib/util/http/index.ts (hunk 2 above)
Comment (rarkins) [reply lib/util/http/index.ts:58]: Let's revert it then and make whatever changes necessary to get tests to pas s

**Pull request merged**